pkg/identity/identitysync/okta.go

```diff
@@ -19,12 +20,14 @@ type OktaSync struct {
 	client   *okta.Client
 	orgURL   gconfig.StringValue
 	apiToken gconfig.SecretStringValue
+	groups   gconfig.StringValue
 }
 
 func (s *OktaSync) Config() gconfig.Config {
 	return gconfig.Config{
 		gconfig.StringField("orgUrl", &s.orgURL, "the Okta organization URL"),
 		gconfig.SecretStringField("apiToken", &s.apiToken, "the Okta API token", gconfig.WithNoArgs("/granted/secrets/identity/okta/token")),
+		gconfig.StringField("groups", &s.groups, "Groups that users are synced from."),
```

Comment:
IMO, we shouldn't break the previous interface/contract, and make this optional.

Now I am not sure if `gconfig.StringField` makes this mandatory, but there is a suspiciously named `gconfig.OptionalStringField()`!

Reply:
My second comment, shall we call this plainly `groups` or something more explicit like `user_groups`?

it is possible later we add a `groups` to refer to the groups we do want to sync.

Reply:
Fixed, now it's optional.

Reply:
> it is possible later we add a groups to refer to the groups we do want to sync.

what do you mean? currently these are the groups that we want to sync users from, right?